pointmatcher/DataPointsFilters/MaxPointCount.cpp

```diff
@@ -66,35 +67,20 @@ MaxPointCountDataPointsFilter<T>::filter(const DataPoints& input)
 template<typename T>
 void MaxPointCountDataPointsFilter<T>::inPlaceFilter(DataPoints& cloud)
 {
-	const size_t N = static_cast<size_t> (cloud.features.cols() - 1);
-	
-	if (maxCount <= N) 
+	const auto N = static_cast<size_t>(cloud.features.cols() - 1);
+
+	if (maxCount <= N)
 	{
-		//Re-init seed at each call, to ensure same results
-		std::srand(seed);
-		
-		for(size_t j=0; j<maxCount; ++j)
+		std::default_random_engine randomNumberGenerator(seed);
```

Comment:
What is the default random engine? Did you evaluate it against other options?

Reply:
In the [documentation](https://en.cppreference.com/w/cpp/numeric/random), it says that the `default_random_engine` is implementation-defined and on my computer (`gcc` 7.5.0) it says that it's `std::minstd_rand0`.

I did some benchmark to evaluate the different ways random numbers can be generated:

![int_vs_real_minstd_rand0](https://user-images.githubusercontent.com/37111793/124630220-36055080-de50-11eb-8394-0520e13e2a7d.png)

![int_vs_real_minstd_rand](https://user-images.githubusercontent.com/37111793/124630731-b4fa8900-de50-11eb-8cce-85b37b8eca1f.png)

![real_minimal_standard_vs_mersenne_twister_double](https://user-images.githubusercontent.com/37111793/124631056-01de5f80-de51-11eb-90ea-959917430657.png)

![real_minimal_standard_vs_mersenne_twister_float](https://user-images.githubusercontent.com/37111793/124631471-68637d80-de51-11eb-9a6a-af6412613c44.png)

From that, we can clearly see that only modifying the old algorithm to use the C++11 `random` library is faster than using C `rand` function, and it also produce better results at shuffling `DataPoints`.